packages/angular/ssr/src/routes/route-config.ts

```diff
@@ -17,30 +17,40 @@ export enum RenderMode {
   AppShell,
 
   /** Server-Side Rendering (SSR) mode, where content is rendered on the server for each request. */
-  SSR,
+  Server,
 
   /** Client-Side Rendering (CSR) mode, where content is rendered on the client side in the browser. */
-  CSR,
+  Client,
 
   /** Static Site Generation (SSG) mode, where content is pre-rendered at build time and served as static files. */
-  SSG,
+  Prerender,
 }
 
 /**
- * Fallback strategies for Static Site Generation (SSG) routes.
+ * Defines the fallback strategies for Static Site Generation (SSG) routes when a pre-rendered path is not available.
+ * This is particularly relevant for routes with parameterized URLs where some paths might not be pre-rendered at build time.
+ *
  * @developerPreview
  */
-export enum SSGFallback {
-  /** Use Server-Side Rendering (SSR) as the fallback for this route. */
-  SSR,
+export enum PrerenderFallback {
+  /**
+   * Fallback to Server-Side Rendering (SSR) if the pre-rendered path is not available.
+   * This strategy dynamically generates the page on the server at request time.
+   */
+  Server,
 
-  /** Use Client-Side Rendering (CSR) as the fallback for this route. */
-  CSR,
+  /**
+   * Fallback to Client-Side Rendering (CSR) if the pre-rendered path is not available.
+   * This strategy allows the page to be rendered on the client side.
+   */
+  Client,
 
-  /** No fallback; Angular will not handle the response if the path is not pre-rendered. */
+  /**
+   * No fallback; if the path is not pre-rendered, the server will not handle the request.
+   * This means the application will not provide any response for paths that are not pre-rendered.
```

Comment:
It's a bit unclear whether that means 404/500 error codes or something else. What would be the behavior in this case if there was no prerendered HTML page found for this request?

Reply:
Currently, setting it to 'None' means Angular will not handle the request and it will instead fallback to the next middleware. 

While we could be more explicit by returning a 404 error ourselves, we currently lack a way to define and serve a custom error page.

Maybe for now, we could send a page with some `404 not found text` and 404 status code?

Reply:
Discussed offline, the behavior makes sense, we'll update the docs to make it more clear.